### PR TITLE
campaigns: patch changeset IDs when migrating

### PR DESCRIPTION
**Pull request by LawnGnome**
In #13001, I added some "evil" to move campaigns back to their original IDs after everything was created. In my rush to do so, I committed one of the basic mistakes of evil-doing: not considering the broader effects of my evil; namely, that changesets have an effective (but unenforced) foreign key on the campaign in their JSON campaign_ids blobs.

Obviously, the solution here is more evil, in the form of an UPDATE query that patches the JSON en masse. You're welcome?

cc: @marekweb, as we'll need this in 3.19 as well.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Looks good, although I must confess the Json operators don’t say anything to me without reading up the docs 🤷‍♂️

**Comment by LawnGnome** [comment]
> Looks good, although I must confess the Json operators don’t say anything to me without reading up the docs man_shrugging

Yeah, they're ugly as (bar `-`, which at least sort of makes sense). I certainly can't remember them.

File: enterprise/internal/campaigns/spec_migration.go [hunk 1]
```diff
@@ -219,6 +219,18 @@ func (svc *Service) MigratePreSpecCampaigns(ctx context.Context) (err error) {
 		// copied the table exactly but didn't recreate campaigns or its ID
 		// sequence, which means we know the sequence is already past the ID
 		// that was on the old campaign.
+		//
```
Comment (mrnugget) [inline enterprise/internal/campaigns/spec_migration.go:222]: (I can't comment on unrelated lines, but: do we need to change `store.Done(err)` further up into `err = store.Done(err)`?

**Review by mrnugget — APPROVED** [review]

File: enterprise/internal/campaigns/spec_migration.go (hunk 1 above)
Comment (LawnGnome) [reply enterprise/internal/campaigns/spec_migration.go:238]: Yes. Added in b2d4497.

**Pull request merged**